ui/lib/core/addon/components/search-select-with-modal.js

```diff
@@ -153,64 +163,62 @@ export default class SearchSelectWithModal extends Component {
       results.unshift(suggestion);
     }
   }
+
   filter(options, searchText) {
     const matcher = (option, text) => defaultMatcher(option.searchText, text);
     return filterOptions(options || [], searchText, matcher);
   }
   // -----
-
   @action
   discardSelection(selected) {
     this.selectedOptions.removeObject(selected);
-    this.allOptions.pushObject(selected);
+    this.dropdownOptions.pushObject(selected);
     this.handleChange();
   }
+
   // ----- adapted from ember-power-select-with-create
   @action
-  searchAndSuggest(term, select) {
+  searchAndSuggest(term) {
     if (term.length === 0) {
-      return this.allOptions;
+      return this.dropdownOptions;
     }
-    if (this.search) {
-      return resolve(this.search(term, select)).then((results) => {
-        if (results.toArray) {
-          results = results.toArray();
-        }
-        this.addCreateOption(term, results);
-        return results;
-      });
+    if (this.args.models?.some((model) => model.includes('policy'))) {
+      term = term.toLowerCase();
     }
-    const newOptions = this.filter(this.allOptions, term);
+    const newOptions = this.filter(this.dropdownOptions, term);
     this.addCreateOption(term, newOptions);
     return newOptions;
   }
+
   @action
-  async selectOrCreate(selection) {
-    // if creating we call handleChange in the resetModal action to ensure the model is valid and successfully created
-    // before adding it to the DOM (and parent model)
-    // if just selecting, then we handleChange immediately
+  selectOrCreate(selection) {
     if (selection && selection.__isSuggestion__) {
+      // user has clicked to create a new item
+      // wait to handleChange below in resetModal
       const name = selection.__value__;
       this.showModal = true;
-      const createRecord = await this.store.createRecord(this.args.model);
-      createRecord.name = name;
-      this.newModelRecord = createRecord;
+      this.args.onCreate({ name });
+      // if firing onCreate does not generate a formModel, then pass name to form component
```

Comment:
Should we instead assume/insist that `onCreate` does generate a formModel? I feel like opening up the component to multiple ways of using it is asking for more logic sprawl